jmx-scraper/src/main/java/io/opentelemetry/contrib/jmxscraper/JmxScraper.java

```diff
@@ -43,14 +52,7 @@ public static void main(String[] args) {
       JmxScraper jmxScraper = new JmxScraper(config);
       jmxScraper.start();
 
-      Runtime.getRuntime()
-          .addShutdownHook(
-              new Thread() {
-                @Override
-                public void run() {
-                  jmxScraper.shutdown();
-                }
-              });
+      Runtime.getRuntime().addShutdownHook(new Thread(jmxScraper::shutdown));
```

Comment:
because the thread is managed within the `JmxMetricInsight` we might even be able to get rid of the executor service, alternatively we could also modify the implementation of `JmxMetricInsight` to provide the ability to use an executor service that we own.